### ISSUE TITLE
Can't get specific error info from mg_connect_client, mg_connect_websocket_client, etc.

### ISSUE THREAD

**Issue by snej** [civetweb/civetweb#456]
The API calls that make connections return error info as a string and nothing else. This makes it difficult or impossible for the program that called the API to know exactly why a connection failed. That's bad for a couple of reasons:

* Some clients [including mine] will retry if the error could be transient, like the connection being dropped. This is actually very important for my library.
* Clients that are themselves libraries [like mine] probably have an API that communicates error codes or specific exceptions to their clients. In this circumstance the only error they can return is something generic like `err_sorry_something_went_wrong_on_the_network`.
* Applications can't present localized (or less-geeky) error messages to the user.

**Comment by bel2125** [comment]
In principle I agree to this approach for new components. However, since civetweb is already in use in many products that are in production state, we need to take care of compatibility as well.

I just saw, you already prepared a pull request. Thank you for your contribution.
Still, for all changes affecting the API, I would prefer we discuss it first for a couple of days, before merging.

**Comment by bel2125** [comment]
Unfortunately your pull request #458 is an incompatible API change, so I cannot merge it.
Please let's discuss the requirements and consequences for the interface here first.

**Comment by bel2125** [comment]
In principle, I agree it would be useful to have return values as well, but we must not change released interfaces. There are already some proposed changes to the client API (see https://github.com/civetweb/civetweb/blob/master/docs/Interface_Changes_1.10.md):

- `mg_download` ... is already superseded by the `mg_connect_client` family and remains as a convenience wrapper
- `mg_connect_client` ... already identified as a insufficient for https, and missing the opportunity to add options (#434)
- `mg_connect_client_secure` ... https enhancement, still missing options (#434)
- `mg_connect_websocket_client` ... also not complete for https and missing options (#434)
- a new function should be more generic and allow any type of yet unknown options (like `mg_start` allows to add new options without changing the interface). ***This additional function could return an error code as well***
- `mg_get_response` ... parameters are sufficient, `mg_get_response_info` as replacement for `mg_get_request_info` for client connections planned. Currently there is no other need for adding an error code output, but it could be stored in the connection and a new `mg_get_error(conn)` function could do that.
- the server functions (in particular `mg_start`) still do not return an error code. A complete solution should work for server and client.

**Comment by snej** [comment]
I'm fine with making the API changes in new functions. I made my changes because I need error codes immediately, for my project's purposes. I can re-sync with your branch after you create the new functions with error codes.

**Comment by bel2125** [comment]
You currently only need it for client functions, right?

**Comment by snej** [comment]
Yes, it's more important for clients. At some point it would be good for the server side too; I'm thinking of P2P cases where both parties would like to know why they couldn't connect with each other. But that's in the future.

**Comment by bel2125** [comment]
Must be handled with #479

**Comment by bel2125** [comment]
duplicate #479
